src/Campaigns/Routes/Campaign.php

```diff
@@ -0,0 +1,249 @@
+<?php
+
+namespace Give\Campaigns\Routes;
+
+use Exception;
+use Give\API\RestRoute;
+use Give\Campaigns\Models\Campaign as CampaignModel;
+use Give\Campaigns\ValueObjects\CampaignStatus;
+use WP_Error;
+use WP_REST_Request;
+use WP_REST_Response;
+use WP_REST_Server;
+
+/**
+ * @unreleased
+ */
+class Campaign implements RestRoute
+{
+
+    /** @var string */
+    protected $endpoint = 'campaigns/(?P<id>[0-9]+)';
+
+    /**
+     * @unreleased
+     */
+    public function registerRoute(): void
+    {
+        // Get campaign
+        register_rest_route(
+            'give-api/v2',
+            $this->endpoint,
+            [
+                [
+                    'methods' => WP_REST_Server::READABLE,
+                    'callback' => [$this, 'handleGetRequest'],
```

Comment:
@alaca I recommend creating a controller to handle the request logic.  This gives a nice separation of the route and request, while giving you a controller to write unit tests for.

You can use the form builder as an example:
https://github.com/impress-org/givewp/blob/8adc13337a1289e1a30f52dc9ec632c9b543d0d9/src/FormBuilder/Routes/RegisterFormBuilderRestRoutes.php#L49

https://github.com/impress-org/givewp/blob/304c495bd0a727b0d8e1149c54e7fb1bfecdbcb0/src/FormBuilder/Controllers/FormBuilderResourceController.php

Reply:
Resolved: https://github.com/impress-org/givewp/pull/7547/commits/309cd0f8e96323c082b2d5eb152a9d7617f76a3c

Reply:
Thanks!

Now that you have a controller, you can even write tests for it! 😄 

Example:
https://github.com/impress-org/givewp/blob/c9b8d7396118eee0cfee485f644b7a26b975214b/tests/Feature/Controllers/FormBuilderResourceControllerTest.php#L21

Reply:
😆 
![image](https://github.com/user-attachments/assets/9b56c9ca-bedf-4842-ac3f-721484ad3b8d)

Resolved: https://github.com/impress-org/givewp/pull/7547/commits/18ee165f25c9cc5722e49da8c5e194387ada5c4e

Reply:
haha Yay!!!